aiida/backends/sqlalchemy/migrations/versions/041a79fc615f_dblog_cleaning.py

```diff
@@ -0,0 +1,182 @@
+                              'NOT LIKE \'aiida.workflows.user.\' || \'%%\')').fetchall()[0][0]
+
+
+def get_serialized_legacy_workflow_logs(connection):
+    """ Get the serialized log records that correspond to legacy workflows """
+    query = connection.execute(
+        'SELECT db_dblog.id, db_dblog.time, db_dblog.loggername, db_dblog.levelname, db_dblog.objpk, '
+        'db_dblog.objname, db_dblog.message, db_dblog.metadata FROM db_dblog WHERE (db_dblog.objname '
+        'LIKE \'aiida.workflows.user.\' || \'%%\')')
+    res = list()
+    for row in query:
+        res.append(dict(list(zip(row.keys(), row))))
+    return dumps_json(res)
+
+
+def get_serialized_unknown_entity_logs(connection):
+    """ Get the serialized log records that correspond to unknown entities """
+    query = connection.execute(
+        'SELECT db_dblog.id, db_dblog.time, db_dblog.loggername, db_dblog.levelname, db_dblog.objpk, '
+        'db_dblog.objname, db_dblog.message, db_dblog.metadata FROM db_dblog WHERE (db_dblog.objname '
+        'NOT LIKE \'node.\' || \'%%\') AND (db_dblog.objname NOT LIKE \'aiida.workflows.user.\' || \'%%\')')
+    res = list()
+    for row in query:
+        res.append(dict(list(zip(row.keys(), row))))
+    return dumps_json(res)
+
+
+def export_and_clean_workflow_logs(connection):
+    """
+    Export the logs records that correspond to legacy workflows and to unknown entities (place them to files
+    and remove them from the DbLog table).
+    """
+    from tempfile import NamedTemporaryFile
+
+    # connection = op.get_bind()
+
+    lwf_no = get_legacy_workflow_log_no(connection)
+    other_no = get_unknown_entity_log_no(connection)
+
+    # If there are no legacy workflow log records or log records of an unknown entity
+    if lwf_no == 0 and other_no == 0:
+        return
+
+    if not settings.TESTING_MODE:
+        click.echo('We found {} log records that correspond to legacy workflows and {} log records to correspond '
+                   'to an unknown entity.'.format(lwf_no, other_no))
+        click.echo(
+            'These records will be removed from the database and exported to JSON files to the current directory).')
+        proceed = click.confirm('Would you like to proceed?', default=True)
+        if not proceed:
+            return
+
+    delete_on_close = False
+    if settings.TESTING_MODE:
+        delete_on_close = True
+
+    # Exporting the legacy workflow log records
+    if lwf_no != 0:
+        # Get the records and write them to file
+        with NamedTemporaryFile(
+                prefix='legagy_wf_logs-', suffix='.log', dir='.', delete=delete_on_close, mode='w+') as handle:
+            # Export the log records
+            handle.write(get_serialized_legacy_workflow_logs(connection))
+        # Now delete the records
+        # connection.execute(
+        connection.execute('DELETE FROM db_dblog WHERE (db_dblog.objname LIKE \'aiida.workflows.user.\' || \'%%\')')
```

Comment:
As in Django, probably you might want to have a utility function generating the `WHERE ...` part of the query for the two queries, and reuse it everywhere.

Reply:
Now I see this. 

Honestly speaking I prefer a few lines of code duplication for a more readable code than having SQL queries being generated half way from different functions to save 3-4 lines of code.

I will do as you propose (to finish this PR as soon as possible) even if believe that the whole migration should be tested on the generated files and not per query (as it was initially written but we failed to do for SQLA).